java/amazon-kinesis-producer-sample/src/com/amazonaws/services/kinesis/producer/sample/SampleProducer.java

```diff
@@ -96,27 +98,34 @@
      * 
      * @see {@link KinesisProducerConfiguration#setRecordTtl(long)}
      */
-    private static final int RECORDS_PER_SECOND = 2000;
+    private static final int RECORDS_PER_SECOND_DEFAULT = 2000;
     
     /**
      * Change this to your stream name.
      */
-    public static final String STREAM_NAME = "test";
+    public static final String STREAM_NAME_DEFAULT = "test";
     
     /**
      * Change this to the region you are using.
      */
-    public static final String REGION = "us-west-1";
+    public static final String REGION_DEFAULT = "us-west-1";
 
     /**
      * Here'll walk through some of the config options and create an instance of
      * KinesisProducer, which will be used to put records.
      *
      * @param region The region of the Kinesis stream being used.
-     * 
+     *
+     * @param connections The max number of connections to use within the KPL to service AWS Requests
+     * @param requestTimeout The configured request timeout for requests to Kinesis
+     * @param bufferTime The configured buffer time for the KPL
+     * @param threadingModel The configured ThreadingModel to use for the KPL. Either "POOLED" or "PER_REQUEST"
+     * @param numberOfThreads When POOLED threading model is used, the number of native threads to run in the
+     *                       Daemon to serve Kinesis requests.
      * @return KinesisProducer instance used to put records.
      */
-    public static KinesisProducer getKinesisProducer(final String region) {
+    public static KinesisProducer getKinesisProducer(final String region, int connections, int requestTimeout,
```

Comment:
It might be cleaner to create a static class to capture the configurations to pass in. It would also avoid a sequence of integer parameters which could confuse a future developer changing this code.

Reply:
Done.